packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts

```diff
@@ -135,30 +155,33 @@ export class PersonalizeMiddleware {
       return response;
     }
 
-    if (personalizeInfo.segments.length === 0) {
+    if (personalizeInfo.variantIds.length === 0) {
       debug.personalize('skipped (no personalization configured)');
       return response;
     }
 
-    // Get segment data from CDP
-    const segmentData = await this.cdpService.getSegments(personalizeInfo.contentId, browserId);
+    // Get variant data from CDP
+    const variantData = await this.cdpService.executeExperience(
+      personalizeInfo.contentId,
+      context,
+      browserId
+    );
     // If a browserId was not passed in (new session), a new browserId will be returned
-    browserId = segmentData.browserId;
-    // This may change, but for now we are only expected to use the first segment if there are multiple
-    segment = segmentData.segments.length ? segmentData.segments[0] : undefined;
+    browserId = variantData.browserId;
+    variantId = variantData.variantId;
```

Comment:
Do we need to reassign `variantId`? I thought that `browserId` only can be generated

Reply:
No, we don't really need this variable declared early on. I've removed it and use the results from `executeExperience` call directly.